plutus-core/untyped-plutus-core/src/UntypedPlutusCore/Evaluation/Machine/Cek/Internal.hs

```diff
@@ -96,6 +101,20 @@ Now clearly 'computeCek' cannot be inlined in 'runCek' whether it's exported or
 Hence we don't export 'computeCek' and instead define 'runCek' in this file and export it, even
 though the rest of the user-facing API (which 'runCek' is a part of) is defined downstream.
 
+Another problem is handling mutual recursion in the 'computeCek'/'returnCek'/'forceEvaluate'/etc
+family. If we keep it on the top level, GHC won't be able to pull the @Ix fun@ constraint out of
+the family (confirmed by inspecting Core: GHC thinks that since the superclass constraints
+populating the dictionary representing the @Ix fun@ constraint are redundant, they can be replaced
+with calls to 'error' in a recursive call, but that changes the dictionary and so it can no longer
+be pulled out of recursion). But that entails passing a redundant argument around, which slows down
+the machine a tiny little bit. Hence we perform the worker-wrapper transformation manually.
+However that allows GHC to inline almost all of the machine into a single definition (with a bunch
+of recursive join points in it), which _seems_ to make the machine slower for some reason. Hence
+we add a NOINLINE pragma to the entering point of the machine to prevent GHC from inlining
+everything, which gives us Core that is basically identical to the one we had before manually
+worker-wrapper transforming the machine, except the dictionary representing the @Ix fun@ constraint
+is not threaded through the machine.
```

Comment:
@kwxm @michaelpj whoever creates the next PR, please also delete these lines.

Reply:
Ah yes, I took your comment and I forgot to update it!